walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,38 +26,73 @@
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
     List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
     Set<String> attendees =  new HashSet<String>();
+    Set<String> optionalAttendees =  new HashSet<String>();
     int meetingDur = (int) request.getDuration();
     attendees.addAll(request.getAttendees());
+    optionalAttendees.addAll(request.getOptionalAttendees());
     if (meetingDur > TimeRange.WHOLE_DAY.duration()){
       return possibleTimes;
     }
-    if (events.isEmpty() || attendees.isEmpty()){
+    if (events.isEmpty()){
       possibleTimes.add(TimeRange.WHOLE_DAY);
       return possibleTimes;
     }
-    ArrayList<TimeRange> busyWithOverlap = getBusyTimes(events, attendees);
-    possibleTimes = findFreeTimes(busyWithOverlap, request.getDuration());
+    possibleTimes = getPossibleTimes(events, attendees, optionalAttendees, request.getDuration());
     return possibleTimes;
   }
 
 /**
-  * Returns an ArrayList of meeting times that are unavailable due to schedule
-  * conflicts and overlapping meeting times within the given group of attendees. 
+  * Returns an ArrayList of possible meeting times that take into account mandatory
+  * attendees & optional attendees schedules.
   *
   * @param  events the Collection of Event objects
-  * @param  requestAttendees the Set of attendees
-  * @return    the ArrayList of all meeting times that do not work with the attendees schedules
+  * @param  requestAttendees the Set of mandatory attendees
+  * @param  optAttendees the Set of optional attendees
+  * @param  duration the length of the requested meeting
+  * @return    the ArrayList of all meeting times that work with the attendees schedules
   */
-  private ArrayList<TimeRange> getBusyTimes(Collection<Event> events, Set<String> requestAttendees) {
+  private ArrayList<TimeRange> getPossibleTimes(Collection<Event> events, Set<String> requestAttendees,
+          Set<String> optAttendees, long duration) {
     ArrayList<TimeRange> busyTimes = new ArrayList<TimeRange>();
+    ArrayList<TimeRange> optionalBusyTimes = new ArrayList<TimeRange>();
     for(Event event : events) {
       Set<String> eventAttendees = event.getAttendees();
       Set<String> overlap = new HashSet<>(eventAttendees);
+      Set<String> overlapOptional = new HashSet<>(eventAttendees);
```

Comment:
overlapOptional is a hard to understand variable. Rename or comment.

Reply:
Fixed